File: src/plugin/command/owner/MemoryUsageCommandPlugin.java
package plugin.command.owner;

import org.redrune.game.entity.actor.player.Player;
import org.redrune.game.content.plugin.type.CommandPlugin;
import org.redrune.utility.functions.Misc;
import plugin.command.CommandManifest;

/**
 * @author Tyluur <itstyluur@gmail.com>
 * @since 9/1/2017
 */
@CommandManifest(description = "Shows how much memory is used.")
public class MemoryUsageCommandPlugin extends CommandPlugin {
	
	@Override
	public void handle(Player player, String[] args, boolean console, boolean clientCommand) {
		String info = Misc.getMemoryUsageInformation();
		System.out.println(info);
		player.getPackets().sendGameMessage(info);
	}
	
	@Override
	public String[] identifiers() {
		return arguments("memused");
	}
}

File: src/main/java/plugin/command/owner/MemoryUsageCommandPlugin.kt
package plugin.command.owner

import game.content.plugin.type.CommandPlugin
import game.entity.actor.player.Player
import utility.functions.Misc
import plugin.command.CommandManifest

/**
 * @author Tyluur <itstyluur@icloud.com>
 * @since 9/1/2017
 */
@CommandManifest(description = "Shows how much memory is used.")
class MemoryUsageCommandPlugin : CommandPlugin() {
    override fun handle(player: Player, args: Array<String>, console: Boolean, clientCommand: Boolean) {
        val info = Misc.getMemoryUsageInformation()
        println(info)
        player.packets.sendMessage(info)
    }

    override fun identifiers(): Array<String> {
        return arguments("memused")
    }
}